src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -110,8 +111,18 @@ public ResponseEntity<Long> getLearningPathId(@PathVariable Long courseId) {
         if (!course.getLearningPathsEnabled()) {
             throw new BadRequestException("Learning paths are not enabled for this course.");
         }
+
+        // generate learning path if missing
         User user = userRepository.getUser();
-        LearningPath learningPath = learningPathRepository.findByCourseIdAndUserIdElseThrow(course.getId(), user.getId());
+        final var learningPathOptional = learningPathRepository.findByCourseIdAndUserId(course.getId(), user.getId());
```

Comment:
How big is this the learning path object? Would it make sense to only retrieve the id from the database, since this is the only thing you return?

Reply:
Other than the id it should only contain the progress (`int`).
Adding a proper query to save on one integer seems overkill. I don't even know if this would actually decrease perforce.